skyportal/handlers/api/catalog_services.py

```diff
@@ -0,0 +1,223 @@
+                                'filter',
+                            }
+                        )
+                        df1.drop(
+                            columns=drop_columns,
+                            inplace=True,
+                        )
+                        dfs.append(df1)
+
+                    if 'PCUL_incbad' in q.lightCurves[transient]:
+                        df2 = q.lightCurves[transient]['PCUL_incbad']
+                        df2.rename(
+                            columns={'Time': 'mjd', 'Rate': 'fluxerr'}, inplace=True
+                        )
+                        drop_columns = list(
+                            set(df2.columns.values)
+                            - {
+                                'mjd',
+                                'ra',
+                                'dec',
+                                'flux',
+                                'fluxerr',
+                                'limiting_mag',
+                                'filter',
+                            }
+                        )
+                        df2.drop(
+                            columns=drop_columns,
+                            inplace=True,
+                        )
+                        dfs.append(df2)
+
+                    df = pd.concat(dfs)
+                    df['ra'] = ra
+                    df['dec'] = dec
+                    df['filter'] = 'swiftxrt'
+                    df['magsys'] = 'ab'
+                    df['zp'] = 0.0
+                    df = df.replace({np.nan: None})
+
+                    data_out = {
+                        'obj_id': obj_id,
+                        'instrument_id': instrument_id,
+                        'group_ids': [g.id for g in user.accessible_groups],
+                        **df.to_dict(orient='list'),
+                    }
+
+                    if len(df.index) > 0:
+                        add_external_photometry(data_out, user)
+                        log(f"Photometry committed to database for {obj_id}")
+                    else:
+                        log(f"No photometry to commit to database for {obj_id}")
+
+                if transient in q.spectra:
+                    filenames = glob.glob(f'{tmpdirname}/{transient}/interval*')
+                    for filename in filenames:
+                        attachment_name = filename.split("/")[-1]
+                        with open(filename, 'rb') as f:
+                            attachment_bytes = base64.b64encode(f.read())
+                        comment = Comment(
+                            text='Swift Detection Spectrum',
+                            obj_id=obj_id,
+                            attachment_bytes=attachment_bytes,
+                            attachment_name=attachment_name,
+                            author=user,
+                            groups=groups,
+                            bot=False,
+                        )
```

Comment:
Is there a reason why we are posting spectra in comments instead of as a Spectrum object?

Reply:
Just because it is an X-ray spectrum :)

Reply:
Not ideal but I imagine starting out a whole infrastructure to handle that sort of data is out of scope for this PR.

Reply:
Yes it would be nice to start building out our capabilities to deal with X-ray / radio data in particular.